### PR TITLE
Add rejection_reason to transfer

### PR DESCRIPTION
**Pull request by MatthewPhinney**

### CODE REVIEW

**Comment by MatthewPhinney** [comment]
The motivation for this PR is that we want to inform clients why the transfer was rejected. The two possibilities are `cancelled

File: schemas/TransferTemplate.json [hunk 1]
```diff
@@ -20,6 +20,11 @@
       "description": "The current state of the transfer (informational only)",
       "$ref": "TransferState.json"
     },
+    "rejected_reason": {
```
Comment (clark800) [inline schemas/TransferTemplate.json]: We already use "cancelled"/"canellation", "executed"/"execution"; should we call it "rejection_reason"? Currently it reads as a "reason that was rejected".

Comment (justmoon) [inline schemas/TransferTemplate.json]: Agreed, I also prefer `rejection_reason` for clarity and consistency.

**Comment by MatthewPhinney** [comment]
@clark800 @justmoon updated, thanks for the feedback.

**Comment by justmoon** [comment]
LGTM

**Comment by clark800** [comment]
LGTM

**Pull request merged**